driver-dse/src/main/java/com/datastax/driver/graph/AbstractGraphStatement.java

```diff
@@ -0,0 +1,141 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.graph;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+import com.google.common.collect.Maps;
+
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.exceptions.InvalidQueryException;
+
+abstract class AbstractGraphStatement<T extends Statement> {
+
+    protected final Map<String, ByteBuffer> payload;
+    protected T wrappedStatement;
+    protected final GraphSession session;
```

Comment:
I'm not sure it's really required to keep a reference to a session; this "ties" the statement to a session, so what happens if it gets executed by a different `Session` instance?
At first glance it seems possible to get rid of this reference:

* It's mainly used to propagate `session.getDefaultGraphOptions()` from prepared to bound statements;
* `getSession()` is never used;
* `GraphStatement` needs the session to instantiate `SimpleStatement`s – but that should change once we rebase on top of current 3.0.